client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx

```diff
@@ -1,53 +1,87 @@
 import React from 'react'
-import { fireEvent, waitFor, render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
 import { BrowserRouter as Router, useParams } from 'react-router-dom'
 import { toast } from 'react-toastify'
 import relativeStages from '../_mocks'
 import Review from './index'
 import * as utilities from '../../../utilities'
 import useAuditSettings from '../../useAuditSettings'
 import { settingsMock, sampleSizeMock } from './_mocks'
-import { IJurisdiction } from '../../useJurisdictions'
 import { contestMocks } from '../Contests/_mocks'
-import { ISampleSizeOption, IContest } from '../../../../types'
 import { jurisdictionMocks } from '../../_mocks'
+import { withMockFetch } from '../../../testUtilities'
 
 const auditSettingsMock = useAuditSettings as jest.Mock
 
-const apiMock: jest.SpyInstance<
-  ReturnType<typeof utilities.api>,
-  Parameters<typeof utilities.api>
-> = jest.spyOn(utilities, 'api').mockImplementation()
-
-const generateApiMock = (
-  sampleSizeReturn: { sampleSizes: ISampleSizeOption[] },
-  contestsReturn: { contests: IContest[] } | Error | { status: 'ok' },
-  jurisdictionReturn:
-    | { jurisdictions: IJurisdiction[] }
-    | Error
-    | { status: 'ok' }
-) => async (endpoint: string) => {
-  switch (endpoint) {
-    case '/election/1/sample-sizes':
-      return sampleSizeReturn
-    case '/election/1/jurisdiction':
-      return jurisdictionReturn
-    case '/election/1/jurisdiction/file':
-      return { file: null, processing: { status: 'PROCESSED' } }
-    case '/election/1/contest':
-      return contestsReturn
-    case '/election/1/round':
-    default:
-      return { status: 'ok' }
-  }
+const apiCalls = {
+  getSampleSizeOptions: {
+    url: '/api/election/1/sample-sizes',
+    response: sampleSizeMock,
+  },
+  putDefaultSampleSize: {
+    url: '/api/election/1/round',
+    response: { status: 'ok' },
+    options: {
+      body: JSON.stringify({
+        sampleSizes: { 'contest-id': 46 },
+        roundNum: 1,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+    },
+  },
+  putNondefaultSampleSize: {
+    url: '/api/election/1/round',
+    response: { status: 'ok' },
+    options: {
+      body: JSON.stringify({
+        sampleSizes: { 'contest-id': 67 },
+        roundNum: 1,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+    },
+  },
+  putCustomSampleSize: {
+    url: '/api/election/1/round',
+    response: { status: 'ok' },
+    options: {
+      body: JSON.stringify({
+        sampleSizes: { 'contest-id': 5 },
+        roundNum: 1,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+    },
+  },
```

Comment:
one pattern i played with is making these API calls functions that take in whatever data you want to customize between tests. e.g. these three could become:

```ts
postRound: (sampleSizes) => ({
    url: '/api/election/1/round',
    response: { status: 'ok' },
    options: {
      body: JSON.stringify({
        sampleSizes,
        roundNum: 1,
      }),
      headers: {
        'Content-Type': 'application/json',
      },
      method: 'POST',
    },
  })
```

and then in the test you can say:
```ts
apiCalls.postRound({ 'contest-id': 46 })
```

dunno if it makes a big difference. curious what you think

Reply:
@jonahkagan Oh yes, I remember that now. Good catch. :slightly_smiling_face: